TTS/vocoder/datasets/wavernn_dataset.py

```diff
@@ -0,0 +1,88 @@
+import torch
+import numpy as np
+from torch.utils.data import Dataset
+
+
+class WaveRNNDataset(Dataset):
+    """
+    WaveRNN Dataset searchs for all the wav files under root path.
+    """
+
+    def __init__(self,
+                 ap,
+                 items,
+                 seq_len,
+                 hop_len,
+                 pad,
+                 mode,
+                 is_training=True,
+                 verbose=False,
+                 ):
+
+        self.ap = ap
+        self.item_list = items
+        self.seq_len = seq_len
+        self.hop_len = hop_len
+        self.pad = pad
+        self.mode = mode
+        self.is_training = is_training
+        self.verbose = verbose
+
+    def __len__(self):
+        return len(self.item_list)
+
+    def __getitem__(self, index):
+        item = self.load_item(index)
+        return item
+
+    def load_item(self, index):
```

Comment:
I think we can compute and load data on the fly. I don't think it is useful to precompute them since it makes it complicates running multiple experiments and on-the-fly data loading has only a negligible overhead. 

What do you think?

Reply:
Yes I would agree, initially I planed to compute them on the fly but skipped it for now.

Reply:
This should be added. As you said the overhead is negligible. I also tried to add feature caching, but it didn't increase the performance as expected.